atom/browser/ui/certificate_trust_mac.mm

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2017 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "atom/browser/ui/certificate_trust.h"
+
+#import <Cocoa/Cocoa.h>
+#import <SecurityInterface/SFCertificateTrustPanel.h>
+
+#include "atom/browser/native_window.h"
+#include "base/strings/sys_string_conversions.h"
+#include "net/cert/cert_database.h"
+
+@interface TrustDelegate : NSObject {
+ @private
+  certificate_trust::ShowTrustCallback callback_;
+  SFCertificateTrustPanel* panel_;
+  scoped_refptr<net::X509Certificate> cert_;
+  SecTrustRef trust_;
+  CFArrayRef cert_chain_;
+  SecPolicyRef sec_policy_;
+}
+
+- (id)initWithCallback:(const certificate_trust::ShowTrustCallback&)callback
+      panel:(SFCertificateTrustPanel*)panel
+      cert:(const scoped_refptr<net::X509Certificate>&)cert
+      trust:(SecTrustRef)trust
+      certChain:(CFArrayRef)certChain
+      secPolicy:(SecPolicyRef)secPolicy;
+
+- (void)panelDidEnd:(NSWindow *)sheet
+        returnCode:(int)returnCode
+        contextInfo:(void*)contextInfo;
+
+@end
+
+@implementation TrustDelegate
+
+- (void)dealloc {
+  [panel_ release];
+  CFRelease(trust_);
+  CFRelease(cert_chain_);
+  CFRelease(sec_policy_);
+
+  [super dealloc];
+}
+
+- (id)initWithCallback:(const certificate_trust::ShowTrustCallback&)callback
+      panel:(SFCertificateTrustPanel*)panel
+      cert:(const scoped_refptr<net::X509Certificate>&)cert
+      trust:(SecTrustRef)trust
+      certChain:(CFArrayRef)certChain
+      secPolicy:(SecPolicyRef)secPolicy {
+  if ((self = [super init])) {
+    callback_ = callback;
+    panel_ = panel;
+    cert_ = cert;
+    trust_ = trust;
+    cert_chain_ = certChain;
+    sec_policy_ = secPolicy;
+  }
+
+  return self;
+}
+
+- (void)panelDidEnd:(NSWindow *)sheet
+        returnCode:(int)returnCode
+        contextInfo:(void*)contextInfo {
+  if (returnCode == NSFileHandlingPanelOKButton) {
+    auto cert_db = net::CertDatabase::GetInstance();
+    // This forces Chromium to reload the certificate since it might be trusted
+    // now.
+    cert_db->NotifyObserversCertDBChanged(cert_.get());
+  }
+
+  callback_.Run(returnCode);
```

Comment:
Since the callback expects a `bool`, what do you think about comparing the value passed to the return code constants for clarity like:

```cpp
 callback_.Run(returnCode == NSFileHandlingPanelOKButton ? true : false);
```

Or maybe add an else block that calls it with `false` and call it with `true` in the if block?